### PR TITLE
Do not upload ghost workspace logs and skip backups

### PR DESCRIPTION
**Pull request by aledbf**
```
no remote storage configured:
    github.com/gitpod-io/gitpod/ws-daemon/pkg/content.(*WorkspaceService).uploadWorkspaceLogs
        github.com/gitpod-io/gitpod/ws-daemon/pkg/content/service.go:552"
```
Internal log https://cloudlogging.app.goo.gl/2ty2pZsFRfVtLARx6

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **aledbf** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)**
- **[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/ws-manager/pkg/manager/monitor.go [hunk 1]
```diff
@@ -800,7 +800,8 @@ func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceOb
 		tpe = api.WorkspaceType_REGULAR
 	}
 
-	doBackup := wso.WasEverReady() && !wso.IsWorkspaceHeadless()
+	doBackup := wso.WasEverReady() && !wso.IsWorkspaceHeadless() && !wsk8s.IsGhostWorkspace(wso.Pod)
```
Comment (csweichel) [inline components/ws-manager/pkg/manager/monitor.go:803]: if `wsk8s.IsGhostWorkspace(wso.Pod) == true` then `wso.IsWorkspaceHeadless() == true`. 


If that's not the case, we have a bug somewhere else.

File: components/ws-manager/pkg/manager/monitor.go [hunk 2]
```diff
@@ -800,7 +800,8 @@ func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceOb
 		tpe = api.WorkspaceType_REGULAR
 	}
 
-	doBackup := wso.WasEverReady() && !wso.IsWorkspaceHeadless()
+	doBackup := wso.WasEverReady() && !wsk8s.IsGhostWorkspace(wso.Pod)
```
Comment (csweichel) [inline components/ws-manager/pkg/manager/monitor.go:803]:
```suggestion
	doBackup := wso.WasEverReady() && !wso.IsWorkspaceHeadless()
```

Comment (aledbf) [reply components/ws-manager/pkg/manager/monitor.go]: done

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5161#issuecomment-897754043" title="Approved">csweichel</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/5161#issuecomment-897754043" title="Approved">csweichel</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)~~ [csweichel]
- ~~[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**